From 154f7357de8b5346b1b78759f4d5d52aa2202acf Mon Sep 17 00:00:00 2001
From: mukesh agrawal <quiche@chromium.org>
Date: Fri, 24 Aug 2012 10:31:18 -0700
Subject: [PATCH] CHROMIUM: wpa_supplicant: disable the SessionTicket extension
 in SSL

Some RADIUS servers choke if the Client Hello message includes
unrecognized options. Turn off the SessionTicket extension, to
work with these servers.

BUG=chromium-os:33832
TEST=WiFiSecMat.072CheckWPA_1x_PEAP
TEST=WiFiSecMat.073CheckWPA_1x_TTLS
TEST=manual (see below)

Manual testing: do packet capture on DUT while associating
with an 802.1x network, verify that the "SessionTicket TLS"
extension is not sent in Client Hello message.

Change-Id: I44027ca1c60004d3eb7df4e9d7050bbdf4476373
Reviewed-on: https://gerrit.chromium.org/gerrit/31382
Reviewed-by: Paul Stewart <pstew@chromium.org>
Commit-Ready: mukesh agrawal <quiche@chromium.org>
Tested-by: mukesh agrawal <quiche@chromium.org>
---
 src/crypto/tls_openssl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/crypto/tls_openssl.c b/src/crypto/tls_openssl.c
index e153422..5b0366f 100644
--- a/src/crypto/tls_openssl.c
+++ b/src/crypto/tls_openssl.c
@@ -1048,6 +1048,8 @@ struct tls_connection * tls_connection_init(void *ssl_ctx)
 #ifdef SSL_OP_NO_COMPRESSION
 	options |= SSL_OP_NO_COMPRESSION;
 #endif /* SSL_OP_NO_COMPRESSION */
+        options |= SSL_OP_NO_TICKET;
+        options |= SSL_OP_NO_SESSION_RESUMPTION_ON_RENEGOTIATION;
 	SSL_set_options(conn->ssl, options);
 
 	conn->ssl_in = BIO_new(BIO_s_mem());
-- 
2.2.0.rc0.207.ga3a616c

